### PR TITLE
Logs agent version on startup

### PR DESCRIPTION
**Pull request by josejulio**
On a Hawkular javaagent it looks like this:
```
14:18:22,592 INFO  ...: Hawkular Agent: Javaagent version 1.0.0.Final-SNAPSHOT
```

On a Hawkular wildfly agent it looks like this:
```
14:11:21,450 INFO  ...: Hawkular Agent: WildFly Agent Subsystem version 1.0.0.Final-SNAPSHOT
```

### CODE REVIEW

**Comment by josejulio** [comment]
@jmazzitelli is this what you wanted?

File: hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/util/Util.java [hunk 1]
```diff
@@ -389,4 +392,17 @@ public static String extractDockerContainerIdFromString(String s) {
...
+        String filename = clazz.getName().replace('.', '/') + ".class";
+        String classPath = clazz.getClassLoader().getResource(filename).toString();
+        if (!classPath.startsWith("jar")) {
+            return null;
+        }
+        String manifestPath = classPath.substring(0, classPath.lastIndexOf('!') + 1) + File.separator + JarFile.MANIFEST_NAME;
+        InputStream stream = new URL(manifestPath).openStream();
+        Manifest manifest = new Manifest(stream);
+        stream.close();
+        return manifest;
```
Comment (jmazzitelli) [inline hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/util/Util.java]: This seems overkill. Can we just remove this Util method and in AgentCoreEngine simply do:

```
log.infoTypeAndVersion(this.getClass().getPackage().getImplementationTitle(),
    this.getClass().getPackage().getImplementationVersion());
```

File: hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/service/AgentCoreEngine.java [hunk 2]
```diff
@@ -225,6 +227,14 @@ public void startHawkularAgent(AgentCoreEngineConfiguration newConfiguration) {
...
 
         try {
+            Manifest agentManifest = Util.manifestFromClass(this.getClass());
+            if (agentManifest != null) {
+                Attributes manifestAttributes = agentManifest.getMainAttributes();
+                String agentType = manifestAttributes.getValue("Implementation-Title");
+                String agentVersion = manifestAttributes.getValue("Implementation-Version");
+                log.infoTypeAndVersion(agentType, agentVersion);
+            }
+
```
Comment (jmazzitelli) [inline hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/service/AgentCoreEngine.java]: Can we just remove all of this and replace it with one line like this:

```
log.infoTypeAndVersion(this.getClass().getPackage().getImplementationTitle(),
    this.getClass().getPackage().getImplementationVersion());
```

**Comment by jmazzitelli** [comment]
@josejulio I'm not sure if getPackage() returns the manifest information - but I think it does. I haven't tested it, but I'm pretty sure I have done something like this in the past.

File: hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/util/Util.java (hunk 1 above)
Comment (josejulio) [reply hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/util/Util.java]: Sure. I didn't know about those.

**Comment by josejulio** [comment]
I'll test and update this.

**Comment by josejulio** [comment]
Just updated it.
Tested on hawkular-javaagent and hawkular-wildfly-agent.

**Pull request merged**